pkg/spoke/hubclientcert/controller.go

```diff
@@ -241,14 +244,15 @@ func (c *ClientCertForHubController) syncCSR(secret *corev1.Secret) (map[string]
 	// create a kubeconfig with references to the key/cert files in kubeconfigSecret if it dose not exists.
 	// So other components deployed in separated deployments are able to access this kubeconfig for hub as
 	// well by sharing the secret
-	if _, ok := secret.Data[KubeconfigFile]; !ok {
+	kubeconfigData, ok := secret.Data[KubeconfigFile]
```

Comment:
this change is used to fix the current bug, for recovery from invalid hub kubeconfig test, the old code will miss the `KubeconfigFile` in hub kubeconfig secret, because
- at the first sync, controller creates the `KubeconfigFile` with an invalid kubeconfig and save it to hub kubeconfig secret 
- at the second sync (recover from the invalid hub kubecofing), because the hub kubeconfig secret already has the `KubeconfigFile`, the `KubeconfigFile` will not be added to `data`, the controller will use the `data` to update hub kubeconfig secret, so the `KubeconfigFile` will miss in hub kubeconfig secret